source/common/formatter/substitution_formatter.cc

```diff
@@ -62,18 +62,19 @@ const std::string SubstitutionFormatUtils::DEFAULT_FORMAT =
     "\"%REQ(:AUTHORITY)%\" \"%UPSTREAM_HOST%\"\n";
 
 FormatterPtr SubstitutionFormatUtils::defaultSubstitutionFormatter() {
-  return FormatterPtr{new FormatterImpl(DEFAULT_FORMAT)};
+  return FormatterPtr{new FormatterImpl(DEFAULT_FORMAT, false)};
 }
 
-const std::string&
+const absl::optional<std::string>
 SubstitutionFormatUtils::protocolToString(const absl::optional<Http::Protocol>& protocol) {
+  absl::optional<std::string> result;
   if (protocol) {
-    return Http::Utility::getProtocolString(protocol.value());
+    result = Http::Utility::getProtocolString(protocol.value());
   }
-  return UnspecifiedValueString;
+  return result;
```

Comment:
In general, instead of:
```
absl::optional<std::string> result;
if (condition) {
  result = some_string;
}
return result;
```
you can just write
```
if (condition) {
  return some_string;
}
return absl::null_opt;
```

Reply:
Left this pattern in a few places where otherwise compiler got confused about a lambda return type, but removed overall.